Mathlib/CategoryTheory/Functor/KanExtension/Preserves.lean

```diff
@@ -0,0 +1,430 @@
+/-
+Copyright (c) 2025 Robin Carlier. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Robin Carlier
+-/
+import Mathlib.CategoryTheory.Functor.KanExtension.Adjunction
+import Mathlib.CategoryTheory.Limits.Preserves.Basic
+
+/-!
+# Preservation of Kan extensions
+
+Given functors `F : A ⥤ B`, `L : B ⥤ C`, and `G : B ⥤ D`,
+we introduce a typeclass `G.PreservesLeftKanExtension F L`, which encodes the fact that
+the left/right Kan extension of `F` along `L` is preserved by the functor `G`.
+
+When the Kan extension is pointwise, It suffices that `G` preserves (co)limits of the relevant
+diagrams.
+
+-/
+
+namespace CategoryTheory.Functor
+
+variable {A B C D : Type*} [Category A] [Category B] [Category C] [Category D]
+  (G : B ⥤ D) (F : A ⥤ B) (L : A ⥤ C)
+
+section LeftKanExtension
+
+/-- `G.PreservesLeftKanExtension F L` asserts that `G` preserves all left kan extensions
+of `F` along `L`. -/
+class PreservesLeftKanExtension where
+  preserves : ∀ (F' : C ⥤ B) (α : F ⟶ L ⋙ F') [IsLeftKanExtension F' α],
+    IsLeftKanExtension (F' ⋙ G) <| whiskerRight α G ≫ (Functor.associator _ _ _).hom
+
+/-- Alternative constructor for `PreservesLeftKanExtension`, phrased in terms of
+`LeftExtension.IsUniversal` instead. -/
+lemma PreservesLeftKanExtension.mk'
+    (preserves : ∀ {E : LeftExtension L F}, E.IsUniversal →
+      Nonempty (LeftExtension.postcompose₂ L F G|>.obj E).IsUniversal) :
+    G.PreservesLeftKanExtension F L where
```

Comment:
There should also be a constructor which would take as an input *one* `E : LeftExtension L F` that is universal and remains so after postcomposing with `G`. (Similarly as for the preservation of limits, we have `preservesColimit_of_preserves_colimit_cocone`.)